lib/clientv2/client.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv2
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	"github.com/projectcalico/libcalico-go/lib/backend"
+	bapi "github.com/projectcalico/libcalico-go/lib/backend/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"github.com/projectcalico/libcalico-go/lib/ipam"
+	"github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/projectcalico/libcalico-go/lib/options"
+)
+
+// client contains
+type client struct {
+	// The backend client is currently public to allow access to datastore
+	// specific functions that are used by calico/node.  This is a temporary
+	// measure and users of the client API should not assume that the backend
+	// will be available in the future.
+	Backend bapi.Client
+
+	// The untyped client used internally
+	untyped untyped
```

Comment:
not too concerned but I think we should rename this to something else

Reply:
Ok, I've made the naming resource-centric rather than "untyped"